system/system_error_monitor/src/system_error_monitor_core.cpp

```diff
@@ -111,9 +111,15 @@ diagnostic_msgs::msg::DiagnosticArray convertHazardStatusToDiagnosticArray(
     diag_array.status.push_back(decorateDiag(hazard_diag, "[Safe Fault]"));
   }
   for (const auto & hazard_diag : hazard_status.diag_latent_fault) {
+    RCLCPP_WARN_THROTTLE(
+      rclcpp::get_logger("system_error_monitor"), *clock, 5000, "[Latent Fault] %s: %s",
+      hazard_diag.name.c_str(), hazard_diag.message.c_str());
     diag_array.status.push_back(decorateDiag(hazard_diag, "[Latent Fault]"));
   }
   for (const auto & hazard_diag : hazard_status.diag_single_point_fault) {
+    RCLCPP_ERROR_THROTTLE(
+      rclcpp::get_logger("system_error_monitor"), *clock, 5000, "[Single Point Fault] %s: %s",
+      hazard_diag.name.c_str(), hazard_diag.message.c_str());
```

Comment:
Since there are cascading failures, I guess the root cause might not appear in the log with `THROTTLE`.

Is it okay?
Also, could you test this behavior? You can use `dummy_diag_publisher`.

Reply:
@kenji-miyake 
I've tested some cases, but I'll test more with `dummy_diag_publisher`.
https://github.com/autowarefoundation/autoware.universe/pull/510#issuecomment-1065316513

I wanted to show the root(?) cause, but lane_departure_checker just outputs `Error`. `vehicle is out of the lane` is better, which I guess is the root cause.
 I confirmed that without THROTTLE, I can output `vehilce is out of the lane`. I really appreciate your comment.

I'm not familiar with cascading failure structure, but is there any way to output just the root cause with THROTTLE?

Reply:
Mm... I meant you have to trigger two errors at the same time.
But in this case, a different diagnostic (the parent of your target diagnostic) was coincidentally displayed, it might be okay.

> I'm not familiar with cascading failure structure, but is there any way to output just the root cause with THROTTLE?

I remember it was difficult.
We have to name each `THROTTLE` a different name.

Reply:
I changed the name of the logger of RCLCPP_ERROR_THROTTLE for the parent and cause reason, but only the parent reason is output. If there seems no good way to output the root reason, I give up for now.

> Mm... I meant you have to trigger two errors at the same time.
Ah, you mean I should try two different errors at the same time. OK, I will.

Reply:
I remember, with the current implementation of `THROTTLE`, you have to decide the names during the build phase, not the runtime phase.

Reply:
Please confirm the source of the macros.

Reply:
Or perhaps I've implemented a hacky way in `ad_service_state_monitor`? :thinking:

Reply:
I've found out it.
https://github.com/tier4/AutowareArchitectureProposal.iv/blob/30f2ca0a82e836e2d787264326af6cf6c832d4b2/system/autoware_state_monitor/include/autoware_state_monitor/rosconsole_wrapper.h#L28

Reply:
I don't understand the macro of THROTTLE

Thank you. Then I will port logThrottleNamed in another PR.

Reply:
@kenji-miyake 
Finally I could implement throttle log output considering the logger name. For now I put them in the same file as follows.

https://github.com/autowarefoundation/autoware.universe/blob/eeb8406ad10c170943a919c35bdb930742eb76d2/system/system_error_monitor/src/system_error_monitor_core.cpp#L29-L48

Reply:
It works well (I put the screenshot in the bottom). I don't use dummy_diag_publisher though.